prowler/providers/azure/services/containerregistry/containerregistry_service.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+
+from azure.mgmt.containerregistry import ContainerRegistryManagementClient
+
+from prowler.lib.logger import logger
+from prowler.providers.azure.azure_provider import AzureProvider
+from prowler.providers.azure.lib.service.service import AzureService
+from prowler.providers.azure.services.monitor.monitor_client import monitor_client
+from prowler.providers.azure.services.monitor.monitor_service import DiagnosticSetting
+
+
+########################## Container Registry
+class ContainerRegistry(AzureService):
+    def __init__(self, provider: AzureProvider):
+        super().__init__(ContainerRegistryManagementClient, provider)
+        self.registries = self._get_container_registries(provider)
+
+    def _get_container_registries(self, provider):
+        logger.info("Container Registry - Getting registries...")
+        registries = {}
+        for subscription, client in self.clients.items():
+            try:
+                registries.update({subscription: []})
```

Comment:
Could you store the registries in a dict where the key is the ID of the registry, you can maintain the ID in the class too.
```suggestion
                registries.update({subscription: {}})
```

Reply:
You mean like this?

```diff
        for subscription, client in self.clients.items():
            try:
+              registries.update({subscription: {}})
                registries_list = client.registries.list()
                for registry in registries_list:
                    resource_group = self._get_resource_group(registry.id)
+                  registries[registry.id].append(
                        ContainerRegistryInfo(
                            id=getattr(registry, "id", ""),
                            name=getattr(registry, "name", ""),
                            location=getattr(registry, "location", ""),
                            resource_group=resource_group,
                            sku=getattr(registry.sku, "name", ""),
                            login_server=getattr(registry, "login_server", ""),
                            public_network_access=getattr(
                                registry, "public_network_access", ""
                            ),
                            admin_user_enabled=getattr(
                                registry, "admin_user_enabled", ""
                            ),
                            monitor_diagnostic_settings=self._get_registry_monitor_settings(
                                registry.name, resource_group, subscription
                            ),
                        )
                    )
```

Reply:
Yes, but dicts does not have `append` method, so I think it will be something more like:

`registries[registry.id] = ContainerRegistryInfo(...)`

You can use the **AKS** service as reference that is done in that way.

Reply:
Ok, my bad. I am not a professional python dev ;).

Please take a look at my changes.